src/main/java/net/mcreator/ui/modgui/TabGUI.java

```diff
@@ -107,22 +136,54 @@ public TabGUI(MCreator mcreator, ModElement modElement, boolean editingMode) {
 			String readableNameFromModElement = StringUtils.machineToReadableName(modElement.getName());
 			name.setText(readableNameFromModElement);
 		}
+		enableBackgroundTexture();
+	}
+
+	private void enableBackgroundTexture() {
+		if(showSearch.isSelected())
+			bgTexture.setEnabled(false);
+		else
+			bgTexture.setEnabled(true);
 	}
 
+	@Override public void reloadDataLists() {
+		super.reloadDataLists();
+
+		ComboBoxUtil.updateComboBoxContents(bgTexture, ListUtils.merge(Collections.singleton(""),
+				mcreator.getFolderManager().getOtherTexturesList().stream().map(File::getName)
+						.collect(Collectors.toList())), "");
+	}
+
+
 	@Override protected AggregatedValidationResult validatePage(int page) {
 		return new AggregatedValidationResult(page1group);
 	}
 
 	@Override public void openInEditingMode(Tab tab) {
+		//Executed for Minecraft 1.16.4 and before
+		//Delete the old texture file if a new one is selected
+		if (Float.parseFloat(org.apache.commons.lang3.StringUtils

```

Comment:
This "hack" should not be here, check where eg. paintings do this. I also suggest copying this in all cases, instead of resorting to the "parsing" hack right now, 1.15.2 will soon be obsolete anyways and additional file does not hurt that much.